tt_eager/tt_dnn/op_library/sdpa/kernels/dataflow/writer_decode_reducer_interleaved.cpp

```diff
@@ -0,0 +1,148 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "dataflow_api.h"
+#include "tt_eager/tt_dnn/kernels/dataflow/generate_bcast_scalar.hpp"
+#include "tt_eager/tt_dnn/kernels/dataflow/generate_reduce_scaler.hpp"
+
+#include "debug/dprint.h"
+
+template<uint32_t tile_bytes, uint32_t num_readers>
+constexpr uint32_t get_barrier_read_threshold() {
+    return ((512 / num_readers) * (1024 + 128)) / tile_bytes;
+}
+
+void kernel_main() {
+    constexpr uint32_t B = get_compile_time_arg_val(0);  // batch size
+    constexpr uint32_t PNHt = get_compile_time_arg_val(1);  // padded number of heads in tiles
+    constexpr uint32_t PSt = get_compile_time_arg_val(2);  // padded layer length in tiles
+    constexpr uint32_t St = get_compile_time_arg_val(3);  // full sequence length of kv cache in tiles
+    constexpr uint32_t DHt = get_compile_time_arg_val(4);  // head dim
+    constexpr uint32_t identity_scalar_packed = get_compile_time_arg_val(5);
+    constexpr uint32_t scale_val = get_compile_time_arg_val(6);
+    constexpr uint32_t num_cores_per_batch = get_compile_time_arg_val(7);  // num cores per batch
+    constexpr uint32_t num_cores = get_compile_time_arg_val(8);  // num running cores in total
+    constexpr uint32_t in0_receiver_semaphore_addr   = get_compile_time_arg_val(9);  // semaphore for reciever
+    constexpr uint32_t cur_batch = get_compile_time_arg_val(10);
+    constexpr uint32_t k_num_chunks = get_compile_time_arg_val(11);  // number of chunks in K, where k_num_chunks*Sk_chunk_t = PSt
+    constexpr uint32_t k_chunk_start = get_compile_time_arg_val(12);
+    constexpr uint32_t k_chunk_end = get_compile_time_arg_val(13);
+    constexpr bool is_out_sharded = get_compile_time_arg_val(14);
```

Comment:
Can remove "interleaved" from the file name

Reply:
removed